reporting/src/main/java/org/wso2/testgrid/reporting/EmailReportGenerator.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.testgrid.reporting;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.Product;
+import org.wso2.testgrid.common.Status;
+import org.wso2.testgrid.common.TestPlan;
+import org.wso2.testgrid.common.config.ConfigurationContext;
+import org.wso2.testgrid.common.config.PropertyFileReader;
+import org.wso2.testgrid.common.exception.TestGridException;
+import org.wso2.testgrid.common.util.TestGridUtil;
+import org.wso2.testgrid.dao.uow.TestPlanUOW;
+import org.wso2.testgrid.reporting.model.email.TestPlanResultSection;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.testgrid.common.TestGridConstants.HTML_LINE_SEPARATOR;
+import static org.wso2.testgrid.common.TestGridConstants.TEST_PLANS_URI;
+
+/**
+ * This class is responsible for generating all the content for the email-report for TestReportEngine.
+ * The report will consist of base details such as product status, git build details, as well as per test-plan
+ * (per infra-combination) details such as test-plan log, test-plan infra combination.
+ */
+public class EmailReportGenerator {
```

Comment:
I have expected a method name 'generate' that contains the email report generation logic.. :)